File: server_dotnetcore/Utils/Commons.cs
using Newtonsoft.Json;

namespace PokeIpsum
{
    public static class Commons
    {
        public static async Task<T> TratarRespostaAPI<T>(HttpResponseMessage resposta)
        {
            if (resposta.IsSuccessStatusCode)
            {
                var conteudo = await resposta.Content.ReadAsStringAsync();
                var resultado = JsonConvert.DeserializeObject<T>(conteudo);

                if (resultado == null)
                {
                    throw new ArgumentNullException(nameof(resultado), $"The PokéAPI response returned no data.");
                }

                return resultado;
            }
            else
            {
                throw new HttpRequestException($"Request to the PokéAPI failed. Status: {resposta.StatusCode}. Message: {resposta.ReasonPhrase}");
            }
        }
    }
}

File: web/utils/commons.js
import axios from 'axios';

async function tratarRespostaAPI(url) {
    try {
        const response = await axios.get(url);
        if (response.status === 200) {
            if (!response.data) {
                throw new Error('The PokéAPI response returned no data.');
            }
            return response.data;
        } else {
            throw new Error(`Request to the PokéAPI failed. Status: ${response.status}. Message: ${response.statusText}`);
        }
    } catch (error) {
        throw new Error(`HTTP Request Failed: ${error.message}`);
    }
}

function extrairNomes(pokemonData) {
    return pokemonData.map(p => p.pokemon ? p.pokemon.name : p.name);
}

export { tratarRespostaAPI, extrairNomes };